vmm/src/device_manager.rs

```diff
@@ -3779,6 +3779,10 @@ impl Aml for DeviceManager {
                     start_of_device_area,
                     end_of_device_area,
                 ),
+                #[cfg(target_arch = "x86_64")]
+                &aml::AddressSpace::new_io(0u16, 0x0cf7u16),
+                #[cfg(target_arch = "x86_64")]
+                &aml::AddressSpace::new_io(0x0d00u16, 0xffffu16),
```

Comment:
I am curious is the I/O ranges for the root bus is hardcoded in the kernel, so that it is always like this?

```
[    0.317757] pci_bus 0000:00: resource 4 [io  0x0000-0x0cf7 window]
[    0.319035] pci_bus 0000:00: resource 5 [io  0x0d00-0xffff window]
```

Reply:
What do you mean hardcoded? This is read from ACPI directly.

Reply:
The only reason why we have a hole from `0x0cf8` to `0x0cff` is because the PCI bus can be accessed through the ports I/O `0x0cf8` and `0x0cfc`.
Since we don't use any other port I/O for any other purpose, we can reserve the rest for PCI I/O BARs.
@rbradford might be worth mentioning this in a comment, don't you think?

Reply:
I don't think a comment would add anything. This function has to be read with the ACPI spec to really understand it anyway and I think it's quite obvious what it does.